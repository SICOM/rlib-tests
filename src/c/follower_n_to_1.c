/*
 *  Copyright (C) 2016 SICOM Systems, INC.
 *
 *  Author: Zoltán Böszörményi <zboszormenyi@sicom.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of version 2 of the GNU General Public
 * License as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program; if not, write to the
 * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

#include <config.h>

#include <stdio.h>
#include <rlib.h>

char *data[7][5] = {
	[0][0] = "first_name",
	[0][1] = "last_name",
	[0][2] = "color",
	[0][3] = "group",
	[0][4] = "breakfast",

	[1][0] = "Bob",
	[1][1] = "Doan",
	[1][2] = "blue",
	[1][3] = "1",
	[1][4] = "Green Eggs And Spam I Am I Am",

	[2][0] = "Terry",
	[2][1] = "Doan",
	[2][2] = "blue",
	[2][3] = "1",
	[2][4] = "Ham and Eggs",

	[3][0] = "Eric",
	[3][1] = "Buruschkin",
	[3][2] = "green",
	[3][3] = "1",
	[3][4] = "Green Eggs And Spam I Am I Am",

	[4][0] = "Mike",
	[4][1] = "Roth",
	[4][2] = "yellow",
	[4][3] = "2",
	[4][4] = "Green Eggs And Spam I Am I Am",

	[5][0] = "Bob",
	[5][1] = "Kratz",
	[5][2] = "pink",
	[5][3] = "2",
	[5][4] = "Green Eggs And Spam I Am I Am",

	[6][0] = "Steve",
	[6][1] = "Tilden",
	[6][2] = "purple",
	[6][3] = "2",
	[6][4] = "Dude",
};

char *more_data[4][3] = {
	[0][0] = "first_name",
	[0][1] = "last_name",
	[0][2] = "initials",

	[1][0] = "Bob",
	[1][1] = "Doan",
	[1][2] = "WRD",

	[2][0] = "Terry",
	[2][1] = "Doan",
	[2][2] = "TD",

	[3][0] = "Eric",
	[3][1] = "Buruschkin",
	[3][2] = "ERB",
};

int main(int argc, char **argv) {
	rlib *r;

	if (argc == 1) {
		printf("usage: %s [ pdf | xml | txt | csv | html ]\n", argv[0]);
		return 1;
	}

	r = rlib_init();
	rlib_add_datasource_array(r, "local_array");
	rlib_add_query_array_as(r, "local_array", data, 7, 5, "data");
	rlib_add_query_array_as(r, "local_array", more_data, 4, 3, "more_data");
	rlib_add_resultset_follower_n_to_1(r, "data", "data.last_name", "more_data", "more_data.last_name");
	rlib_add_report(r, "follower.xml");
	rlib_set_output_format_from_text(r, argv[1]);
	rlib_execute(r);
	rlib_spool(r);
	rlib_free(r);

	return 0;
}
