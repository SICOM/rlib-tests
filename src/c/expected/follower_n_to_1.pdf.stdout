%PDF-1.3
%·¾­ª
1 0 obj
<<
/Type /Catalog
/Pages 2 0 R
>>
endobj
2 0 obj
<<
/Type /Pages
/Kids [ 4 0 R ]
/Count 1
>>
endobj
3 0 obj
<<
/Producer (Haru Free PDF Library 2.2.1)
/Title (RLIB Report)
>>
endobj
4 0 obj
<<
/Type /Page
/MediaBox [ 0 0 612 792 ]
/Contents 5 0 R
/Resources <<
/ProcSet [ /PDF /Text /ImageB /ImageC /ImageI ]
/Font <<
/F1 7 0 R
/F2 8 0 R
/F3 9 0 R
/F4 10 0 R
>>
>>
/Parent 2 0 R
/Rotate 270
/Annots [ 11 0 R 12 0 R 13 0 R 14 0 R 15 0 R 16 0 R 17 0 R 18 0 R 19 0 R 20 0 R 21 0 R 22 0 R 23 0 R 24 0 R 25 0 R 26 0 R ]
>>
endobj
5 0 obj
<<
/Length 6 0 R
>>
stream
1 0 0 1 0 792 cm
0 -1 1 0 0 0 cm
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 16.56 Tm
(Page:   1) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 11 Tf
BT
1 0 0 1 14.4 589.23999 Tm
(Test Report \050Normal\051 AND THE REPORT FORMAT IS: PDF       3    ) Tj
ET
/F1 9 Tf
/F1 11 Tf
/F2 11 Tf
BT
1 0 0 1 14.4 578.23999 Tm
(Test Report \050Bold\051) Tj
ET
/F1 11 Tf
/F1 9 Tf
/F1 11 Tf
/F3 11 Tf
BT
1 0 0 1 14.4 567.23999 Tm
(Test Report \050Italics\051) Tj
ET
/F1 11 Tf
/F1 9 Tf
/F1 11 Tf
/F2 11 Tf
/F4 11 Tf
BT
1 0 0 1 14.4 556.23999 Tm
(Test Report \050Bold and Italics\051) Tj
ET
/F3 11 Tf
/F1 11 Tf
/F1 9 Tf
/F1 11 Tf
/F2 11 Tf
/F4 11 Tf
BT
1 0 0 1 14.4 545.23999 Tm
(BOO: ) Tj
ET
/F3 11 Tf
/F1 11 Tf
/F1 9 Tf
/F1 11 Tf
BT
1 0 0 1 47.4 545.23999 Tm
(mike is a ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
0 0 0 rg
0 0 0 RG
14.4 537.59998 763.20001 0.856 re
f
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
14.4 528.59998 334.79999 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 530.76001 Tm
(First Name          ) Tj
ET
BT
1 0 0 1 122.4 530.76001 Tm
( Last Name           ) Tj
ET
BT
1 0 0 1 235.8 530.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 530.76001 Tm
(Color               ) Tj
ET
14.4 527.59998 763.20001 0.856 re
f
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 514.59998 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 516.76001 Tm
(Bob                 ) Tj
ET
BT
1 0 0 1 122.4 516.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 514.59998 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 516.76001 Tm
(Doan                ) Tj
ET
BT
1 0 0 1 235.8 516.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 516.76001 Tm
(blue                 WRD   MATCH    MATCH    MATCH   ) Tj
ET
BT
1 0 0 1 527.40002 516.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 516.76001 Tm
(MATCH   ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 505.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 507.76001 Tm
(Bob                 ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
122.4 505.60001 5.4 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 122.4 507.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 505.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 507.76001 Tm
(Doan                ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
235.8 505.60001 340.20001 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 235.8 507.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 507.76001 Tm
(blue                 TD    NO_MATCH NO_MATCH NO_MATCH) Tj
ET
BT
1 0 0 1 527.40002 507.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 507.76001 Tm
(NO_MATCH) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 496.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 498.76001 Tm
(Terry               ) Tj
ET
BT
1 0 0 1 122.4 498.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 496.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 498.76001 Tm
(Doan                ) Tj
ET
BT
1 0 0 1 235.8 498.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 498.76001 Tm
(blue                 WRD   NO_MATCH NO_MATCH NO_MATCH) Tj
ET
BT
1 0 0 1 527.40002 498.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 498.76001 Tm
(NO_MATCH) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 487.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 489.76001 Tm
(Terry               ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
122.4 487.60001 5.4 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 122.4 489.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 487.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 489.76001 Tm
(Doan                ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
235.8 487.60001 340.20001 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 235.8 489.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 489.76001 Tm
(blue                 TD    MATCH    MATCH    MATCH   ) Tj
ET
BT
1 0 0 1 527.40002 489.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 489.76001 Tm
(MATCH   ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 478.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 480.76001 Tm
(Eric                ) Tj
ET
BT
1 0 0 1 122.4 480.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 478.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 480.76001 Tm
(Buruschkin          ) Tj
ET
BT
1 0 0 1 235.8 480.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 480.76001 Tm
(green                ERB   MATCH    MATCH    MATCH   ) Tj
ET
BT
1 0 0 1 527.40002 480.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 480.76001 Tm
(MATCH   ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 469.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 471.76001 Tm
(Mike                ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
122.4 469.60001 5.4 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 122.4 471.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 469.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 471.76001 Tm
(Roth                ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
235.8 469.60001 340.20001 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 235.8 471.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 471.76001 Tm
(yellow                     NO_MATCH NO_MATCH NO_MATCH) Tj
ET
BT
1 0 0 1 527.40002 471.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 471.76001 Tm
(NO_MATCH) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 460.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 462.76001 Tm
(Bob                 ) Tj
ET
BT
1 0 0 1 122.4 462.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 460.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 462.76001 Tm
(Kratz               ) Tj
ET
BT
1 0 0 1 235.8 462.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 462.76001 Tm
(pink                       NO_MATCH NO_MATCH NO_MATCH) Tj
ET
BT
1 0 0 1 527.40002 462.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 462.76001 Tm
(NO_MATCH) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
1 1 0 rg
1 1 0 RG
14.4 451.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 14.4 453.76001 Tm
(Steve               ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
122.4 451.60001 5.4 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 122.4 453.76001 Tm
( ) Tj
ET
1 1 0 rg
1 1 0 RG
127.8 451.60001 108 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 127.8 453.76001 Tm
(Tilden              ) Tj
ET
0.89804 0.89804 0.89804 rg
0.89804 0.89804 0.89804 RG
235.8 451.60001 340.20001 8.856 re
f
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 235.8 453.76001 Tm
( ) Tj
ET
BT
1 0 0 1 241.2 453.76001 Tm
(purple                     NO_MATCH NO_MATCH NO_MATCH) Tj
ET
BT
1 0 0 1 527.40002 453.76001 Tm
( ) Tj
ET
BT
1 0 0 1 532.79999 453.76001 Tm
(NO_MATCH) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf

endstream
endobj
6 0 obj
6701
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Courier
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
8 0 obj
<<
/Type /Font
/BaseFont /Courier-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
9 0 obj
<<
/Type /Font
/BaseFont /Courier-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Courier-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
11 0 obj
<<
/Rect [ 514.59998 669.59998 523.59998 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
12 0 obj
<<
/Rect [ 514.59998 556.20001 523.59998 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
13 0 obj
<<
/Rect [ 505.60001 669.59998 514.59998 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
14 0 obj
<<
/Rect [ 505.60001 556.20001 514.59998 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
15 0 obj
<<
/Rect [ 496.60001 669.59998 505.60001 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
16 0 obj
<<
/Rect [ 496.60001 556.20001 505.60001 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
17 0 obj
<<
/Rect [ 487.60001 669.59998 496.60001 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
18 0 obj
<<
/Rect [ 487.60001 556.20001 496.60001 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
19 0 obj
<<
/Rect [ 478.60001 669.59998 487.60001 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
20 0 obj
<<
/Rect [ 478.60001 556.20001 487.60001 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
21 0 obj
<<
/Rect [ 469.60001 669.59998 478.60001 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
22 0 obj
<<
/Rect [ 469.60001 556.20001 478.60001 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
23 0 obj
<<
/Rect [ 460.60001 669.59998 469.60001 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
24 0 obj
<<
/Rect [ 460.60001 556.20001 469.60001 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
25 0 obj
<<
/Rect [ 451.60001 669.59998 460.60001 777.59998 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
26 0 obj
<<
/Rect [ 451.60001 556.20001 460.60001 664.20001 ]
/Type /Annot
/Subtype /Link
/A <<
/Type /Action
/S /URI
/URI (http:\057\057www.yahoo.com)
>>
/P 4 0 R
>>
endobj
xref
0 27
0000000000 65535 f
0000000015 00000 n
0000000064 00000 n
0000000123 00000 n
0000000205 00000 n
0000000548 00000 n
0000007303 00000 n
0000007323 00000 n
0000007418 00000 n
0000007518 00000 n
0000007621 00000 n
0000007729 00000 n
0000007903 00000 n
0000008077 00000 n
0000008251 00000 n
0000008425 00000 n
0000008599 00000 n
0000008773 00000 n
0000008947 00000 n
0000009121 00000 n
0000009295 00000 n
0000009469 00000 n
0000009643 00000 n
0000009817 00000 n
0000009991 00000 n
0000010165 00000 n
0000010339 00000 n
trailer
<<
/Root 1 0 R
/Info 3 0 R
/Size 27
>>
startxref
10513
%%EOF
